### PR TITLE
docs: document behaviour when things go wrong and redo k8s howto

### PR DESCRIPTION
**Pull request by JohannesMessner**
**ToDO**
- [x] document error handling behaviour. See here: https://docs-failures--jina-docs.netlify.app/fundamentals/flow/when-things-go-wrong/
- [x] polish k8s HowTo. See here: https://docs-failures--jina-docs.netlify.app/how-to/kubernetes/
- [x] @samsja edit the "expose flow" details

### CODE REVIEW

**Comment by samsja** [comment]
I am wondering if it could be useful to have a section where we explain what could have gone wrong when user see `|Gateway: Communication error with deployment...`

Could be nice to have the error message in our docs so that user will find this page when googling this error. (our with docsQA)

**Comment by JohannesMessner** [comment]
@samsja please add your suggested improvements to the snippets that access the deployed Flow!

File: docs/how-to/kubernetes.md [hunk 1]
```diff
@@ -1,197 +1,122 @@
...
 
-By *simple* in this context we mean a Flow without replicated or sharded Executors - you will see how to use those in
-Kubernetes {ref}`later on <kubernetes-replicas>`.
+A `Flow` is composed of `Executors` which run Python code
+defined to operate on a `DocumentArray`. These Executors live in different runtimes depending on how you want to deploy
+your Flow. 
 
-For now, define a Flow.
-You can either do this through the Flow {ref}`YAML interface <flow-yaml-spec>` or directly in Python, like we do here:
+When deployed in Kubernetes, these Executors will run inside Kubernetes Pods as containers and their lifetime will be handled
```
Comment (JoanFM) [inline docs/how-to/kubernetes.md:10]:
```suggestion
When deployed in Kubernetes, these Executors will run inside Kubernetes Pods as containers and their lifetime will be orchestrated
```

File: docs/how-to/kubernetes.md [hunk 2]
```diff
@@ -201,76 +126,116 @@ f = (
     .add(name='encoder', uses='jinahub+docker://CLIPEncoder', replicas=2)
     .add(
         name='indexer',
-        uses='jinahub+docker://ANNLiteIndexer',
+        uses='jinahub+docker://PQLiteIndexer',
```
Comment (JoanFM) [inline docs/how-to/kubernetes.md:129]: ANNLite is the new name, so keep ANNLite

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: docs/how-to/kubernetes.md [hunk 3]
```diff
@@ -178,36 +177,65 @@ kubectl get pods -n custom-namespace
 NAME                              READY   STATUS    RESTARTS   AGE
 encoder-8b5575cb9-bh2x8           1/1     Running   0          60m
 encoder-8b5575cb9-gx78g           1/1     Running   0          60m
-encoder-head-55bbb477ff-p2bmk   1/1     Running   0          60m
+encoder-head-55bbb477ff-p2bmk     1/1     Running   0          60m
```
Comment (JoanFM) [inline docs/how-to/kubernetes.md:180]: we should rerun this to show new message without `encoder-head`

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Review by JoanFM — APPROVED** [review]

File: docs/how-to/kubernetes.md [hunk 4]
```diff
@@ -178,36 +177,64 @@ kubectl get pods -n custom-namespace
...
+indexer-head-6fcc679d95-8mrm6     1/1     Running   0          60m
 ```
 
 Note that the Jina gateway was deployed with name `gateway-7df8765bd9-xf5tf`.
 
 Once we see that all the Deployments in the Flow are ready, we can start indexing documents.
 
-```python
-import portforward
+We need first to forward the Kubernetes services to your local machine by doing
```
Comment (JohannesMessner) [inline docs/how-to/kubernetes.md:190]:
```suggestion
First, you need to forward the Kubernetes services to your local machine by doing
```

File: docs/how-to/kubernetes.md [hunk 5]
```diff
@@ -178,36 +177,64 @@ kubectl get pods -n custom-namespace
...
+
+```bash 
+kubectl port-forward svc/gateway 8080:8080
+```
+
+Now we can define a client to connect to our Flow running in Kubernetes:
```
Comment (JohannesMessner) [inline docs/how-to/kubernetes.md:196]:
```suggestion
Now you can define a Client to connect to your Flow running in Kubernetes:
```

File: docs/how-to/kubernetes.md [hunk 6]
```diff
@@ -178,36 +177,64 @@ kubectl get pods -n custom-namespace
...
 
+```python
 from jina.clients import Client
+
+client = Client(host='http://localhost:8080')
+```
+
+Then let's index the set of images we want to search:
```
Comment (JohannesMessner) [inline docs/how-to/kubernetes.md:204]:
```suggestion
Then you can index the set of images that you want to search:
```

File: docs/how-to/kubernetes.md [hunk 7]
```diff
@@ -178,36 +177,64 @@ kubectl get pods -n custom-namespace
...
+print(f'Indexed Documents: {len(indexed_docs)}')
 ```
 
+```shell
+Indexed Documents: 99
+```
+
+We indexer 99 Documents !
+
+Then let's search for the closest image to our query image:
```
Comment (JohannesMessner) [inline docs/how-to/kubernetes.md:224]:
```suggestion
Then, search for the closest image to our query image:
```

**Review by samsja — APPROVED** [review]

**Pull request merged**